feat(frontend): add login iframe route

nickhsine opened the pull request:
## Summary: 
Introduce a dedicated login route that renders the external signin widget inside an iframe while wiring through the environment configuration needed by the frontend.

## Changes:
- update .env.example with the new NEXT_PUBLIC_LOGIN_WIDGET_URL and align NEXT_PUBLIC_LOGIN_URL with the signin route.
- expose loginWidgetUrl via src/environment-variables.ts so code can reference the iframe target.
- add src/app/(sticky-header)/login/page.tsx, which renders the login widget iframe and preserves any destination query parameter.

## Additional Notes:
The iframe defaults to https://accounts.twreporter.org/signin-widget when the env var isn’t set, and the login route now defaults to /login.
 
## Related PR(s): 
Both this PR and twreporter/membership-fe#1162 add iframe-friendly signin flows. Our frontend exposes the widget URL + /login route while the membership repo adds the signinWidget/registerWidget routes and widget logic that strips layout and preserves destinations.

## Reference(s):
https://www.notion.so/twreporter/Accounts-2738dbdca93280bc9ed7e52043683701

nickhsine committed:
feat(frontend): add login iframe route

- add NEXT_PUBLIC_LOGIN_WIDGET_URL and adjust NEXT_PUBLIC_LOGIN_URL in .env.example so the app can build the iframe target and default signin path
- expose loginWidgetUrl from src/environment-variables.ts alongside the existing loginUrl
- add (sticky-header)/login/page.tsx to render the login widget as a full-height iframe, passing along any destination query parameter

nickhsine commented on packages/frontend/src/environment-variables.ts:
```diff
@@ -11,7 +11,10 @@ const searchEngineID = process.env.SEARCH_ENGINE_ID || ''
+const loginUrl = process.env.NEXT_PUBLIC_LOGIN_URL || '/signin'
+const loginWidgetUrl =
+  process.env.NEXT_PUBLIC_LOGIN_WIDGET_URL ||
+  'https://accounts.twreporter.org/signin'
```
fixed

nickhsine commented on packages/frontend/src/app/(sticky-header)/login/page.tsx:
```diff
@@ -0,0 +1,14 @@
+  }
+}) {
+  const iframeSrc = searchParams.destination
+    ? `${envVars.loginWidgetUrl}?destination=${searchParams.destination}`
```
fixed.

ericsen-tsai commented on packages/frontend/src/app/(sticky-header)/login/page.tsx:
```diff
@@ -0,0 +1,17 @@
+
+  const iframeSrc = `${envVars.loginWidgetUrl}?destination=${encodeURIComponent(destination)}`
+
+  return <iframe style={{ width: '100%', height: '100vh' }} src={iframeSrc} />
```
nitpick:
you can use tailwind classname for consistency

nickhsine replied:
fixed.

ericsen-tsai commented on packages/frontend/.env.example:
```diff
@@ -16,4 +16,7 @@ SEARCH_API_KEY=search-api-key
 # Login Url
-NEXT_PUBLIC_LOGIN_URL=https://staging-accounts.twreporter.org
\ No newline at end of file
+NEXT_PUBLIC_LOGIN_URL=/login
+
+# Login Widget Url
+NEXT_PUBLIC_LOGIN_WIDGET_URL=https://accounts.twreporter.org/signin-widget
```
suggestion:
we may need to modify and add these environment variables to different env files

nickhsine replied:
fixed in [8866a94](https://github.com/kids-reporter/kids-reporter-monorepo/pull/845/commits/8866a9440f651017dc52fa67ec53c45914ed5ac9)

nickhsine committed:
refactor(frontend): sanitize the login destination query param

nickhsine committed:
chore(frontend): expand public env vars for login integration

- add the API gateway endpoint, base URL, standardized release env flag, /login entry, and login widget URL to every .env.{env}.public, keeping the widget URL
  aligned with staging/production hosts (with dev pointing to staging accounts).
- align the ndx env file with the others by introducing the same set of keys rather than only the graphql and login values already present.

nickhsine commented on packages/frontend/src/app/(sticky-header)/login/page.tsx:
```diff
@@ -0,0 +1,14 @@
+  }
+}) {
+  const iframeSrc = searchParams.destination
+    ? `${envVars.loginWidgetUrl}?destination=${searchParams.destination}`
```
fixed in [ddfc2dc](https://github.com/kids-reporter/kids-reporter-monorepo/pull/845/commits/ddfc2dc464cd371f493b746c44598cd9b502530a)

nickhsine commented on packages/frontend/src/app/(sticky-header)/login/page.tsx:
```diff
@@ -0,0 +1,14 @@
+  const iframeSrc = searchParams.destination
+    ? `${envVars.loginWidgetUrl}?destination=${searchParams.destination}`
+    : envVars.loginWidgetUrl
+  return <iframe style={{ width: '100%', height: '100vh' }} src={iframeSrc} />
```
fixed.

nickhsine commented:
I merge this PR for testing first.
Welcome to leave comments if you find anything to change.
